cpp/src/genetic/reg_stack.cuh

```diff
@@ -58,38 +58,20 @@ struct stack {
    *       to push more than `MaxSize` elements leads to all sorts of incorrect
    *       behavior.
    */
-  HDI void push(DataT val) {
-#pragma unroll
-    for (int i = 0; i < MaxSize; ++i) {
-      if (elements_ == i) {
-        regs_[i] = val;
-        ++elements_;
-      }
-    }
-  }
```

Comment:
it's better to stick with the unoptimized for-loop for the sake of the compiler, Vimarsh. Just looping backwards to circumvent the bug would do. By indexing the `regs_[]` array with a runtime variable, we'd incur register spills...